plugins/source/azure/policies/queries/storage/no_public_blob_container.sql

```diff
@@ -7,10 +7,10 @@ SELECT
     azsc.subscription_id                                                        AS subscription_id,
     azsc.id                                                                     AS resrouce_id,
     CASE
-        WHEN azsc.properties->>'publicAccess' = 'None'
-         AND NOT (asa.properties->>'allowBlobPublicAccess')::boolean
-        THEN 'pass'
-        ELSE 'fail'
+        WHEN (asa.properties->>'allowBlobPublicAccess')::BOOLEAN = true 
+        AND (azsc.properties->>'publicAccess') <> 'None' 
```

Comment:
I don't find the CIS Azure 1.3's logic clear.  However, the intent is clear to `Disable anonymous access to blob containers and disallow blob public access on storage account.`

Intent wise, we're looking to ensure that any containers that can thus be public fail the control. 

There are 2 different scenarios here: a classic storage account and the newer (V2) storage account. 

This logic will work for the newer storage account as if allowBlobPublicAccess is disabled on the storage account, containers will have `None` as their properties.  Thus there's no scenario with V2 storage where allowBlobPublicAccess is true and the container publicAccess property is not `None`.  

However, I'm not sure if this covers classic storage accounts since classic storage accounts require the container setting to be configured for each container (https://learn.microsoft.com/en-us/azure/storage/blobs/anonymous-read-access-prevent-classic?tabs=portal)

Snippet from Azure: `Anonymous access to this container is being blocked because anonymous access is disabled on this storage account.`. 

CIS Azure 1.3 snippet below: 

```
Anonymous, public read access to a container and its blobs can be enabled in Azure Blob storage. It grants read-only access to these resources without sharing the account key, and without requiring a shared access signature. It is recommended not to provide anonymous access to blob containers until, and unless, it is strongly desired. A shared access signature token should be used for providing controlled and timed access to blob containers. If no anonymous access is needed on the storage account, it’s recommended to set allowBlobPublicAccess false.
```